spyder_env_manager/spyder/widgets/main_widget.py

```diff
@@ -0,0 +1,408 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Spyder env manager Main Plugin Widget.
+"""
+
+# Third party imports
+from qtpy import PYQT5
+from qtpy.QtCore import QUrl
+from qtpy.QtWidgets import (
+    QInputDialog,
+    QMessageBox,
+    QComboBox,
+    QStackedLayout,
+)
+import os.path as osp
+from string import Template
+import pathlib
+import os
+from qtpy.compat import getopenfilenames
+from spyder_kernels.utils.iofuncs import iofunctions
+
+# Local imports
+from spyder.utils.misc import getcwd_or_home, remove_backslashes
+from spyder.api.translations import get_translation
+from spyder.api.widgets.main_widget import PluginMainWidget
+from spyder.widgets.browser import FrameWebView
+from spyder.utils.palette import QStylePalette
+from spyder.config.base import get_module_source_path
+import qtawesome as qta
+from spyder.utils.icon_manager import ima
+from spyder.utils.conda import get_list_conda_envs_cache
+from spyder.utils.pyenv import get_list_pyenv_envs_cache
+from spyder_env_manager.spyder.widgets.helper_widgets import MessageComboBox
+from spyder_env_manager.spyder.widgets.packages_table import EnvironmentPackagesTable
+
+# Localization
+_ = get_translation("spyder")
+
+
+# =============================================================================
+# ---- Constants
+# =============================================================================
+PLUGINS_PATH = get_module_source_path("spyder", "plugins")
+TEMPLATES_PATH = osp.join(pathlib.Path(__file__), "spyder", "assets", "templates")
+MAIN_BG_COLOR = QStylePalette.COLOR_BACKGROUND_1
+separador = osp.sep
+ENVIRONMENT_MESSAGE = open(
+    osp.join(
+        separador.join(osp.dirname(os.path.abspath(__file__)).split(separador)[:-2]),
+        "spyder",
+        "assets",
+        "templates",
+        "environment_info.html",
+    )
+).read()
+CSS_PATH = osp.join(PLUGINS_PATH, "help", "utils", "static", "css")
```

Comment:
Maybe we should use `pathlib.Path` instead of `os.path` - `osp` in general (to define the constants here but in general handle paths in the code)?